### ISSUE TITLE
"Failed to load System.Private.CoreLib.dll (error code 0x80070002)" for self-contained binary

### ISSUE THREAD

**Issue by anthony-c-martin** [dotnet/runtime#109593]
### Description

When using self-contained binaries, we've run into a strange error message if the binary is executed on a path outside of `/usr/local/bin`. This was reported to us under https://github.com/Azure/bicep/issues/15278.

Here's the error:
```
./bicep build test.bicep    
Failed to load System.Private.CoreLib.dll (error code 0x80070002)
Path: /System.Private.CoreLib.dll
Error message: Could not load file or assembly '/System.Private.CoreLib.dll'. The system cannot find the file specified.
(0x80070002)
```

### Reproduction Steps

1. Clone https://github.com/Azure/bicep
2. Run `dotnet publish --configuration release --self-contained true -r bicep-linux-arm64 ./src/Bicep.Cli/Bicep.Cli.csproj`
3. Copy the binary to a linux arm64 container, and try to run it from a path **outside** of `/usr/local/bin` - e.g. `./bicep build <foo>`

### Expected behavior

The binary runs successfully and returns exit code 0.

### Actual behavior

The following error is returned:

```
./bicep build test.bicep    
Failed to load System.Private.CoreLib.dll (error code 0x80070002)
Path: /System.Private.CoreLib.dll
Error message: Could not load file or assembly '/System.Private.CoreLib.dll'. The system cannot find the file specified.
(0x80070002)
```

### Regression?

Unknown

### Known Workarounds

Copy the binary to `/usr/local/bin` on the system.

**Comment by agocke** [comment]
Our official bits don't support bicep:

>  The specified RuntimeIdentifier 'bicep-linux-arm64' is not recognized.

Is this a source-built copy of the runtime?

**Comment by anthony-c-martin** [comment]
> Our official bits don't support bicep:
> 
> > The specified RuntimeIdentifier 'bicep-linux-arm64' is not recognized.
> 
> Is this a source-built copy of the runtime?

It's a binary we've built using `dotnet` with [self-contained mode](https://learn.microsoft.com/en-us/dotnet/core/deploying/#publish-self-contained).

**Comment by agocke** [comment]
Right, but where are you getting _dotnet_? You can't pass `-r -r bicep-linux-arm64` on an official build of the dotnet SDK.

**Comment by anthony-c-martin** [comment]
🤦 sorry - I had a typo in my original repro. It was meant to be `linux-arm64`, not `bicep-linux-arm64`. I've updated it.

**Comment by janvorli** [comment]
Right, it is pretty weird handling only amd64 that way, it doesn't make any sense. The code is unchanged since the creation of the open source .NET Core (except of the ifdef that used to be `_AMD64_`, but that's just cosmetic).
